### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by JimmyDeemo** [googlesamples/unity-jar-resolver#460]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2019.4.24f1
  * External Dependency Manager version: Latest
  * Source you installed EDM4U: Trying to build from source
  * Features in External Dependency Manager in use: Android resolver, iOS resolver
  * Plugins SDK in use: N/A
  * Platform you are using the Unity editor on: Mac

### [REQUIRED] Please describe the issue here:
I am unable to build from source with a fresh clone of the master branch. I am running MacOS 11.4 and it appears that mono will not execute on this platform. See attached log. Please advise if I am able to build from source using this platform and if not, please can a feature request be added for support?
[building_resolver.log](https://github.com/googlesamples/unity-jar-resolver/files/7046083/building_resolver.log)

#### Please answer the following, if applicable:
What's the issue repro rate? 100%

What happened? How can we make the problem occur?
Fresh clone of the master branch. Navigate to the project directory and execute `./gradlew build --stacktrace --info`. Build fails.

**Comment by JimmyDeemo** [comment]
Seems similat to #448.

**Comment by paulinon** [comment]
Hi @JimmyDeemo,

When utilizing the resolver, you can add this in your project by importing the .unitypackagefile or using the Package Manager. In order to identify what's causing this behavior, could you provide the complete steps in replicating this issue?

**Comment by JimmyDeemo** [comment]
Hello @paulinon 

I'm not trying to utilise the resolver. I am trying to build from source from my own fork so I can attempt a fix for #427 and possibly contribute to the project.

Building from source is referenced here; https://github.com/googlesamples/unity-jar-resolver#building-from-source. It only says to run `.gradlew build` which I have done and have found my setup has an error.

Are there more pre-requisites needed? Can I build from source without mono? Please help me to build from source rather than integrating the package. I would be happy to file this issue in another way if that would be helpful.

**Comment by JimmyDeemo** [comment]
Ah I think I have found what the cause is. For some reason my setup mean that a red '⍉' character is added whenever a command errors;

<img width="1136" alt="image" src="https://user-images.githubusercontent.com/4234718/131495587-f8268821-7924-47e6-91d2-4299b4b87fb1.png">

Looks like this is spilling over in Warp.

**Comment by chkuang-g** [comment]
That is odd.

First of all, could you make sure if `/Applications/Unity/Unity.app/Contents/MonoBleedingEdge/bin/mono` exists?

Second, could you run with `--no-build-cache` and `--no-daemon`?

And yes, doing the following shall do as well.
```
export MONO_EXE=/Applications/Unity/Unity.app/Contents/MonoBleedingEdge/bin/mono
``` 

Sorry for the inconvenience.  I tried to build with `2019.4.28f1` and did not see any error like yours.  I am a bit confused about why it happened as well.

Shawn

**Comment by JimmyDeemo** [comment]
Yes, that path does indeed exist. Note: The version of mono there is different from the one in the 2019 installation;

```
▶ /Applications/Unity/Unity.app/Contents/MonoBleedingEdge/bin/mono -V   
Mono JIT compiler version 5.0.1 (unity-2017.4-mbe/10aeba76d03 Mon Oct 22 21:37:46 GMT 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen (concurrent by default)
                                                                                                     
▶ /Applications/Unity/Hub/Editor/2019.4.24f1/Unity.app/Contents/MonoBleedingEdge/bin/mono -V
Mono JIT compiler version 5.11.0 ((HEAD/90cf2678d79 Fri Feb  5 18:05:48 GMT 2021)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      shared_perfcounters
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          supported, not enabled.
	GC:            sgen (concurrent by default)

```

Running `./gradlew --no-build-cache --no-daemon build` without adding the MONO_EXE export didn't work. I have also tried setting this export with both versions of mono listed above and get a similar error though it does seem to be getting further. I noticed that the gradle.properties file had cached the standard versions too so cleared it such that it would cache the 2019 path but still it didn't build. I have attached logs of both attempts.

Very sorry, I just don't know what is wrong with my setup. Conflicting mono versions perhaps? Gradle issues? I also noticed that the python installed with Homebrew was greater than the version you mentioned, could the build process be using an older version? How would I check that?

[build_2019-mono.log](https://github.com/googlesamples/unity-jar-resolver/files/7089431/build_2019-mono.log)
[build.log](https://github.com/googlesamples/unity-jar-resolver/files/7089432/build.log)

**Comment by chkuang-g** [comment]
I think the better solution is to add CI build test for EDM4U against different version of Unity from different OS.

Will update once we have something like this.
